pkg/collector/adapters/config_to_probe_test.go

```diff
@@ -0,0 +1,243 @@
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrServiceNotAMap, err)
+}
+
+func TestShouldErrorIfNoServiceExtensions(t *testing.T) {
+	configStr := `service:
+  pipelines:
+    traces:
+      receivers: [otlp]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrNoServiceExtensions, err)
+}
+
+func TestShouldErrorIfBadlyFormattedServiceExtensions(t *testing.T) {
+	configStr := `service:
+  extensions:
+    this: should-not-be-a-map`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrServiceExtensionsNotSlice, err)
+}
+
+func TestShouldErrorIfNoHealthCheckInServiceExtensions(t *testing.T) {
+	configStr := `service:
+  extensions: [pprof]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrNoServiceExtensionHealthCheck, err)
+}
+
+func TestShouldErrorIfNoExtensions(t *testing.T) {
+	configStr := `service:
+  extensions: [health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrNoExtensions, err)
+}
+
+func TestShouldErrorIfBadlyFormattedExtensions(t *testing.T) {
+	configStr := `extensions: [hi]
+service:
+  extensions: [health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrExtensionsNotAMap, err)
+}
+
+func TestShouldErrorIfNoHealthCheckExtension(t *testing.T) {
+	configStr := `extensions:
+  pprof:
+service:
+  extensions: [health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrNoExtensionHealthCheck, err)
+}
+
+func TestShouldErrorIfNoHealthCheckExtension_mustMatchFirstHealthCheck(t *testing.T) {
+	configStr := `extensions:
+  health_check:
+service:
+  extensions: [health_check/1, health_check]`
+
+	// prepare
+	config, err := adapters.ConfigFromString(configStr)
+	require.NoError(t, err)
+	require.NotEmpty(t, config)
+
+	// test
+	_, err = adapters.ConfigToContainerProbe(logger, config)
+	assert.Equal(t, adapters.ErrNoExtensionHealthCheck, err)
```

Comment:
I would expect this to not return an error?

Reply:
Agreed, that makes more sense. Have since adjusted logic.